main.py

```
@@ -0,0 +1,116 @@
+from fastapi import FastAPI, Depends, HTTPException, Query
+from sqlalchemy.orm import Session
+from typing import List, Optional
+
+import models
+import schemas
+import crud
+from database import engine, get_db
+
+# Create database tables
+models.Base.metadata.create_all(bind=engine)
+
+# Initialize FastAPI application
+app = FastAPI(
+    title="Library Management API",
+    description="A simple library management system API",
+    version="1.0.0"
+)
+
+
+# Root endpoint
+@app.get("/")
+def read_root():
+    return {
+        "message": "Welcome to Library Management API",
+        "endpoints": {
+            "authors": "/authors/",
+            "books": "/books/",
+            "docs": "/docs"
+        }
+    }
+
+
+# Author Endpoints
+@app.post("/authors/", response_model=schemas.AuthorResponse, status_code=201)
+def create_author(
+        author: schemas.AuthorCreate,
+        db: Session = Depends(get_db)
+):
+    """Create a new author."""
+    try:
+        return crud.create_author(db=db, author=author)
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Error creating author: {str(e)}"
+        )
```

Catching `Exception` broadly hides the real error type. The `Author.name` column has a unique constraint (models.py), so duplicate inserts raise `sqlalchemy.exc.IntegrityError`. Instead of catching all exceptions, catch `IntegrityError` and return a clear 400 message (e.g. 'Author with this name already exists'); let unexpected errors bubble up or return a 500 after logging.